feat(route): 掘金 开发者活动

moke8 opened the pull request:
<!-- 
如有疑问，请参考 https://github.com/DIYgod/RSSHub/discussions/8002
Reference: https://github.com/DIYgod/RSSHub/discussions/8002
-->

## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

<!--
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

Please include route starts with /, with all required and optional parameters. Fail to comply will result in your pull request being closed automatically.
```route
/some/route
/some/other/route
/dont/use/this/or/modify/it
/use/the/fenced/code/block/below
```
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
If your changes are not related to route, please fill in `routes` with `NOROUTE`. Fail to comply will result in your PR being closed.
-->

```routes
/juejin/events/:city?
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x]  跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x]  文档说明 Documentation
  - [x]  中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

TonyRL commented on docs/programming.md:
```diff
@@ -925,6 +925,10 @@ GitHub 官方也提供了一些 RSS:
 
+### 开发者活动
+
+<Route author="moke8" example="/juejin/events/shanghai" path="/juejin/events/:city" :paramsDesc="['活动地点, 可选, 一般为城市名全拼如shanghai, 支持城市可在掘金活动页查阅']" radar="1" rssbud="1"/>
```
```suggestion
<Route author="moke8" example="/juejin/events/shanghai" path="/juejin/events/:city?" :paramsDesc="['活动地点, 可选, 一般为城市名全拼如shanghai, 支持城市可在掘金活动页查阅']" radar="1" rssbud="1"/>
```
Path here must match `maintainer.js` and `router.js`

TonyRL commented on lib/v2/juejin/events.js:
```diff
@@ -0,0 +1,44 @@
+    const items = response.data.data.map((item) => {
+        const content = item.title;
+        const title = content;
+        const guid = item.id;
```
This guid is way too easy for collision
You can use something like
```suggestion
        const guid = `juejin:events:${city ? `${city}:` : ''}${item.id}`;
```
instead

TonyRL commented on lib/v2/juejin/events.js:
```diff
@@ -0,0 +1,44 @@
+        const title = content;
+        const guid = item.id;
+        const link = item.url;
+        const imgs = `<img src="${item.screenshot}">`;
+        const description = `
+            ${content.replace(/\n/g, '<br>')}<br>
+            ${imgs}<br>
+        `;
```
Please use [art-template](https://docs.rsshub.app/joinus/script-standard.html#template)

TonyRL commented on lib/v2/juejin/events.js:
```diff
@@ -0,0 +1,44 @@
+
+    ctx.state.data = {
+        title: `开发者活动 ${cityName}`,
+        link: `https://juejin.cn/events/${encodeURIComponent(city)}`,
```
```suggestion
        link: `https://juejin.cn/events/${city}`,
```
No need to encode plain alphabets

TonyRL commented on lib/v2/juejin/maintainer.js:
```diff
@@ -10,4 +10,5 @@ module.exports = {
     '/shares/:userId': ['qiwihui'],
     '/tag/:tag': ['isheng5'],
     '/trending/:category/:type': ['moaix'],
+    '/events/:city?': ['moke8'],
```
Please insert new entries in alphabetical order

TonyRL commented on lib/v2/juejin/router.js:
```diff
@@ -10,4 +10,5 @@ module.exports = (router) => {
     router.get('/shares/:userId', require('./shares'));
     router.get('/tag/:tag', require('./tag'));
     router.get('/trending/:category/:type', require('./trending'));
+    router.get('/events/:city?', require('./events'));
```
Please insert new entries in alphabetical order

TonyRL commented on lib/v2/juejin/events.js:
```diff
@@ -0,0 +1,44 @@
+        url: 'https://api.juejin.cn/event_api/v1/event/event_list',
+        json: {
+            city: city === 'all' ? '' : city,
+            count: 20,
```
```suggestion
            count: ctx.query.limit ? parseInt(ctx.query.limit) : 20,
```
This way users can specify the number of items they want using common parameters

moke8 commented:
机器人报的问题不是我提交的代码

TonyRL reviewed (commented):
Please take a look at **my** review instead, and remove any changes on `yarn.lock`

moke8 committed:
掘金 开发者活动

moke8 committed:
Merge branch 'DIYgod:master' into master

github-actions[bot] committed:
style: auto format

dependabot[bot] committed:
chore(deps-dev): bump prettier from 2.8.3 to 2.8.4 (#239)

Bumps [prettier](https://github.com/prettier/prettier) from 2.8.3 to 2.8.4.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.8.3...2.8.4)

---
updated-dependencies:
- dependency-name: prettier
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump rand-user-agent from 1.0.104 to 1.0.105 (#244)

Bumps [rand-user-agent](https://github.com/WebScrapingAPI/rand-user-agent) from 1.0.104 to 1.0.105.
- [Release notes](https://github.com/WebScrapingAPI/rand-user-agent/releases)
- [Commits](https://github.com/WebScrapingAPI/rand-user-agent/commits)

---
updated-dependencies:
- dependency-name: rand-user-agent
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump ioredis from 5.3.0 to 5.3.1 (#248)

Bumps [ioredis](https://github.com/luin/ioredis) from 5.3.0 to 5.3.1.
- [Release notes](https://github.com/luin/ioredis/releases)
- [Changelog](https://github.com/luin/ioredis/blob/main/CHANGELOG.md)
- [Commits](https://github.com/luin/ioredis/compare/v5.3.0...v5.3.1)

---
updated-dependencies:
- dependency-name: ioredis
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump rand-user-agent from 1.0.105 to 1.0.106 (#249)

Bumps [rand-user-agent](https://github.com/WebScrapingAPI/rand-user-agent) from 1.0.105 to 1.0.106.
- [Release notes](https://github.com/WebScrapingAPI/rand-user-agent/releases)
- [Commits](https://github.com/WebScrapingAPI/rand-user-agent/commits)

---
updated-dependencies:
- dependency-name: rand-user-agent
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump jest from 29.4.2 to 29.4.3 (#251)

Bumps [jest](https://github.com/facebook/jest/tree/HEAD/packages/jest) from 29.4.2 to 29.4.3.
- [Release notes](https://github.com/facebook/jest/releases)
- [Changelog](https://github.com/facebook/jest/blob/main/CHANGELOG.md)
- [Commits](https://github.com/facebook/jest/commits/v29.4.3/packages/jest)

---
updated-dependencies:
- dependency-name: jest
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump rand-user-agent from 1.0.106 to 1.0.107 (#252)

Bumps [rand-user-agent](https://github.com/WebScrapingAPI/rand-user-agent) from 1.0.106 to 1.0.107.
- [Release notes](https://github.com/WebScrapingAPI/rand-user-agent/releases)
- [Commits](https://github.com/WebScrapingAPI/rand-user-agent/commits)

---
updated-dependencies:
- dependency-name: rand-user-agent
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump rand-user-agent from 1.0.107 to 1.0.108 (#256)

Bumps [rand-user-agent](https://github.com/WebScrapingAPI/rand-user-agent) from 1.0.107 to 1.0.108.
- [Release notes](https://github.com/WebScrapingAPI/rand-user-agent/releases)
- [Commits](https://github.com/WebScrapingAPI/rand-user-agent/commits)

---
updated-dependencies:
- dependency-name: rand-user-agent
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump proxy-chain from 2.2.0 to 2.2.1 (#260)

Bumps [proxy-chain](https://github.com/apify/proxy-chain) from 2.2.0 to 2.2.1.
- [Release notes](https://github.com/apify/proxy-chain/releases)
- [Changelog](https://github.com/apify/proxy-chain/blob/master/CHANGELOG.md)
- [Commits](https://github.com/apify/proxy-chain/compare/v2.2.0...v2.2.1)

---
updated-dependencies:
- dependency-name: proxy-chain
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump @vuepress/plugin-pwa from 1.9.8 to 1.9.9 (#264)

Bumps [@vuepress/plugin-pwa](https://github.com/vuejs/vuepress/tree/HEAD/packages/@vuepress/plugin-pwa) from 1.9.8 to 1.9.9.
- [Release notes](https://github.com/vuejs/vuepress/releases)
- [Changelog](https://github.com/vuejs/vuepress/blob/master/CHANGELOG.md)
- [Commits](https://github.com/vuejs/vuepress/commits/v1.9.9/packages/@vuepress/plugin-pwa)

---
updated-dependencies:
- dependency-name: "@vuepress/plugin-pwa"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

DIYgod commented:
This PR has been open for too long and will be closed first. If you have any questions, feel free to continue the discussion below.
